File: 2 homework/w2a_earth.cpp
// AUTHOR Shen Han shawnhan@bu.edu
// AUTHOR Changlong Jiang cljiang@bu.edu
// AUTHOR Xinyu Li lxinyu@bu.edu

#include <iostream>
#include <math.h>
using namespace std;
double giveElectonNum(double radio,int ifNoNeuton=0){

	double mEarth=5.97e24;
	double mProton=1.6726e-27;
	if (ifNoNeuton==1)
		return (long int)mEarth/mProton;
	else{
		double protonRadio=1/(1/radio+1);
		double cc=round((mEarth*protonRadio)/mProton)/pow(2,43);
		return cc;
	}
}
double estimatedBoundRadio=0.95;
double lowerBoundRadio=0.87;
double upperBoundRadto=1;
int main() {
	double a1=giveElectonNum(estimatedBoundRadio);
	double a2=giveElectonNum(lowerBoundRadio);
	double a3=giveElectonNum(upperBoundRadto);
	cout<<a1<<endl;
	cout<<a2<<endl;
	cout<<a3<<endl;
	return 0;
}

File: 2 homework/w2a_earth.py
# -*- coding: utf-8 -*-
# AUTHOR Shen Han shawnhan@bu.edu
# AUTHOR Changlong Jiang cljiang@bu.edu
# AUTHOR Xinyu Li lxinyu@bu.edu
def giveElectonNum(radio,ifNoNeutron=0):#radio is No
    mEarth=5.9736*(10**24)
    mP=1.6726*(10**(-27))
    if ifNoNeutron==1:
        return mEarth/mP
    else:        
        protonRadio=1/(1/radio+1)
        return (mEarth*protonRadio)/mP

estimatedBoundRadio=0.95
lowerBoundRadio=0.87
upperBoundRadto=1

print(giveElectonNum(estimatedBoundRadio)/(2**43))
print(giveElectonNum(lowerBoundRadio)/(2**43))
print(giveElectonNum(upperBoundRadto)/(2**43))